plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQuerySplitManager.java

```diff
@@ -119,15 +120,18 @@ private static boolean emptyProjectionIsRequired(Optional<List<BigQueryColumnHan
         return projectedColumns.isPresent() && projectedColumns.get().isEmpty();
     }
 
-    private List<BigQuerySplit> readFromBigQuery(ConnectorSession session, TableDefinition.Type type, TableId remoteTableId, Optional<List<BigQueryColumnHandle>> projectedColumns, int actualParallelism, Optional<String> filter)
+    private List<BigQuerySplit> readFromBigQuery(ConnectorSession session, TableDefinition.Type type, TableId remoteTableId, Optional<List<BigQueryColumnHandle>> projectedColumns, int actualParallelism, TupleDomain<ColumnHandle> tableConstraint)
     {
         checkArgument(projectedColumns.isPresent() && projectedColumns.get().size() > 0, "Projected column is empty");
+        Optional<String> filter = BigQueryFilterQueryBuilder.buildFilter(tableConstraint);
 
         log.debug("readFromBigQuery(tableId=%s, projectedColumns=%s, actualParallelism=%s, filter=[%s])", remoteTableId, projectedColumns, actualParallelism, filter);
         List<BigQueryColumnHandle> columns = projectedColumns.get();
-        List<String> projectedColumnsNames = columns.stream()
-                .map(BigQueryColumnHandle::getName)
-                .collect(toImmutableList());
+        ImmutableList.Builder<String> projectedColumnsNames = ImmutableList.builder();
+        projectedColumnsNames.addAll(columns.stream().map(BigQueryColumnHandle::getName).toList());
+        tableConstraint.getDomains().ifPresent(domains -> domains.keySet().stream()
+                .map(column -> ((BigQueryColumnHandle) column).getName())
+                .forEach(projectedColumnsNames::add));
```

Comment:
Otherwise, `TestBigQueryAvroConnectorTest.testEmptyProjectionView` throws the following error:
```
io.grpc.StatusRuntimeException: INVALID_ARGUMENT: request failed: Query error: Unrecognized name: regionkey at [1:105]
```